addons/base_vat/models/res_partner.py

```diff
@@ -704,6 +705,62 @@ def check_vat_t(self, vat):
         if self.country_id.code == 'JP':
             return self.simple_vat_check('jp', vat)
 
+    def check_vat_br(self, vat):
+        '''
+        Example of a Brazilian CNPJ number: 76.634.583/0001-74.
+        The 13th digit is the check digit of the previous 12 digits.
+        The check digit is calculated by multiplying the first 12 digits by weights and calculate modulo 11 of the result.
+        The 14th digit is the check digit of the previous 13 digits. Calculated the same way.
+        Both remainders are appended to the first 12 digits.
+        '''
+        if not vat:
+            return False
+
+        def _l10n_br_calculate_mod_11(check, weights):
```

Comment:
I'm not sure we want to keep the prefix l10n_br (if really needed, we could keep br if we fear conflict name in function, but not l10n at least)